packages/eslint-plugin-pf-codemods/src/rules/v6/chipReplaceWithLabel/chip-replace-with-label.ts

```diff
@@ -0,0 +1,202 @@
+          context
+            .getSourceCode()
+            .getText(
+              badgeAttribute?.value
+                ? parseBadgeAttributeValue(badgeAttribute.value)
+                : undefined
+            )
+        ),
+        fixer.remove(badgeAttribute),
+      ]
+    : [];
+};
+
+const renameOnClickAttribute = (node: JSXElement, fixer: Rule.RuleFixer) => {
+  const onClickAttribute = getJSXAttribute(node, 'onClick');
+
+  return onClickAttribute
+    ? [fixer.replaceText(onClickAttribute.name, 'onClose')]
+    : [];
+};
+
+// https://github.com/patternfly/patternfly-react/pull/10049
+module.exports = {
+  meta: { fixable: 'code' },
+  create: function (context: Rule.RuleContext) {
+    const { imports } = getFromPackage(
+      context,
+      '@patternfly/react-core/deprecated'
+    );
+
+    const chipImport = imports.find(
+      (specifier) => specifier.imported.name === 'Chip'
+    );
+
+    const chipGroupImport = imports.find(
+      (specifier) => specifier.imported.name === 'ChipGroup'
+    );
+
+    return !chipImport && !chipGroupImport
+      ? {}
+      : {
+          ImportDeclaration(node: ImportDeclaration) {
+            context.report({
+              node,
+              message: `Chip has been deprecated. Running the fix flag will replace Chip and ChipGroup components with Label and LabelGroup components respectively.`,
+              fix(fixer) {
+                return [
+                  ...renameImportSpecifier(chipImport, node, fixer, 'Label'),
+                  ...renameImportSpecifier(
+                    chipGroupImport,
+                    node,
+                    fixer,
+                    'LabelGroup'
+                  ),
+                ];
+              },
+            });
+          },
+
+          JSXElement(node: JSXElement) {
+            if (node.openingElement.name.type === 'JSXIdentifier') {
+              if (
+                chipImport &&
+                chipImport.local.name === node.openingElement.name.name
+              ) {
+                context.report({
+                  node,
+                  message: `Chip has been deprecated. Running the fix flag will replace Chip and ChipGroup components with Label and LabelGroup components respectively.`,
+                  fix(fixer) {
+                    return [
+                      fixer.replaceText(node.openingElement.name, 'Label'),
+                      fixer.insertTextAfter(
+                        node.openingElement.name,
+                        ' variant="outline"'
+                      ),
+                      ...renameOnClickAttribute(node, fixer),
+                      ...(node.closingElement
+                        ? [
+                            fixer.replaceText(
+                              node.closingElement.name,
+                              'Label'
+                            ),
+                            ...moveBadgeAttributeToBody(node, fixer, context),
```

Comment:
Is it possible that someone would use a Chip with Badge but without children (self-closing)? I am just wondering, whether this case should be covered:
```
<Chip onClick={onClick} badge={<Badge isRead={true}>7</Badge>} />
```
![Screenshot 2024-03-26 at 16 09 06](https://github.com/patternfly/pf-codemods/assets/84135613/942bcb20-d691-48e4-9ec1-e2125188283e)

I guess probably not, or ... it would be too much work to handle it and it is not very likely.